include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -2799,13 +2799,27 @@ namespace internal
 } // namespace internal
 
 
+template <typename V>
+using is_compressed_op = decltype(std::declval<V>().is_compressed());
```

Comment:
Very close -- would you be willing to move the `is_compressed_op` type into namespace `internal::AffineConstraints`?